### ISSUE TITLE
Social Icons: Preset colors do not stay in sync with Global Styles presets

### ISSUE THREAD

**Issue by kjellr** [WordPress/gutenberg#35480]
When you assign a preset color to the Social Links block, it records the preset's hex color in the markup like so:

```
<!-- wp:social-links {"iconColor":"foreground","iconColorValue":"#000000"} -->
```

If someone were to go and modify the value of that `foreground` preset via Global Styles, the social links block does not update to the new color value. The original hex color defined in the block markup is still used. 

## GIF:

Note that I'm changing the color of "foreground" via Global Styles, but that update is not reflected in the Social Links block color:

![social-links-color](https://user-images.githubusercontent.com/1202812/136615949-c066adad-aaf3-497d-b54f-d2cb313ea8e1.gif)

---
Gutenberg version 11.7.0-rc.1
Noticed during the development of Twenty Twenty-Two: https://github.com/WordPress/twentytwentytwo/pull/74#discussion_r725239866

**Comment by ockham** [comment]
I don't recall the original rationale for the `iconColor`/`iconColorValue` pair, but maybe @aaronrobertshaw can help here? 😄

**Comment by noisysocks** [comment]
`iconColorValue` and `iconBackgroundColorValue` exist as fallback attributes so that the colour remains if you switch to a theme that doesn't have a colour named e.g. `primary`.

https://github.com/WordPress/gutenberg/blob/b1d8f7af02f3c3bb5949efb702d0f3dd215e1185/packages/block-library/src/social-links/edit.js#L172-L174

The reason this bug happens is because Social Links passes the fallback attributes (`iconColorValue` and `iconBackgroundColorValue`) down as context instead of the primary attributes (`iconColor` and `iconBackgroundColor`).

https://github.com/WordPress/gutenberg/blob/b1d8f7af02f3c3bb5949efb702d0f3dd215e1185/packages/block-library/src/social-links/block.json#L36-L40

One fix for this would be to pass down both the primary and fallback attributes so that the Social Link block can set the colour to `iconColor.color || iconColorValue`. 

Another fix could be to remove the fallback attributes altogether and only use the primary attributes. This is what https://github.com/WordPress/gutenberg/pull/29565 does, which should fix this bug.

I'm not sure if the fallback is necessary? Do we do it for other blocks? It would be good if Social Links can use the attributes that setting `supports.color` adds.

@scruffian @oandregal @aaronrobertshaw

**Comment by aaronrobertshaw** [comment]
Sorry I'm late to the conversation on this!

> iconColorValue and iconBackgroundColorValue exist as fallback attributes so that the colour remains if you switch to a theme that doesn't have a colour named e.g. primary.

From memory, the issue regarding the loss of a named color selection when switching themes was raised in the review of the PR [implementing the color options for the social links](https://github.com/WordPress/gutenberg/pull/28084) block. I don't feel strongly about maintaining this behaviour.

> Another fix could be to remove the fallback attributes altogether and only use the primary attributes. This is what #29565 does, which should fix this bug.

The current colors implementation for the social links was added before we had the ability to [skip serialization of block support styles](https://github.com/WordPress/gutenberg/pull/29142). 

There was plenty of discussion on the original PR around applying colors at an individual level vs once only on the primary social links block. It was the desire to have the colors set only once that ultimately drove the context-based approach. Initially, CSS variables were used to pass the colors down however that ran into [some issues with KSES](https://github.com/WordPress/gutenberg/issues/29297). 

This block was a driving force in [evolving the `__experimentalSkipSerialization` feature](https://github.com/WordPress/gutenberg/issues/28913) for block supports. It was intended to be moved over to block supports and https://github.com/WordPress/gutenberg/pull/29565 is still listed in the [tracking issue ](https://github.com/WordPress/gutenberg/issues/28913) to handle that conversion.

> It would be good if Social Links can use the attributes that setting supports.color adds.

Agreed. I'm happy to help out with reviews or pick up https://github.com/WordPress/gutenberg/pull/29565 once 5.9 is released. I was under the impression that particular PR was still waiting on the ability to define custom labels but would need to follow that thread further.

**Comment by donnapep** [comment]
Noting here that this is also an issue when you're trying to update the color of the icons in a style variation. Although `iconColor` is set to `primary`, the color doesn't update because `iconColorValue` is fixed.

**Comment by bradhogan** [comment]
+1 I don't understand why `iconColorVaule` is even an attribute. Or if it's going to be required, we should be able to set the value to something like `var:preset|color|primary` to stay consistent.

**Comment by ndiego** [comment]
This issue is now fixed in #51020
